cupy/core/core.pyx

```diff
@@ -2287,10 +2210,127 @@ cpdef _scatter_op_single(ndarray a, ndarray indices, v, int axis=0, op=''):
     if op == 'update':
         _scatter_update_kernel(
             v, indices, cdim, rdim, adim, a.reduced_view())
+    elif op == 'add':
+        if not issubclass(v.dtype.type, (numpy.int32, numpy.float32)):
+            raise TypeError(
+                'scatter_add only supports int32 and float32 as data type')
+        _scatter_add_kernel(
+            v, indices, cdim, rdim, adim, a.reduced_view())
```

Comment:
I see, but I still recommend you to add the following comment before the ``if`` statement:
```
# give some constraints on types because atomicAdd() in CUDA 7.5 only supports int32, int64, and float32.
```